daft/runners/ray_runner.py

```diff
@@ -412,7 +411,7 @@ def __init__(self, max_task_backlog: int | None) -> None:
         self.reserved_cores = 0
 
         self.threads_by_df: dict[str, threading.Thread] = dict()
-        self.results_by_df: dict[str, Queue] = defaultdict(Queue)
```

Comment:
you can still rely on defaultdict by doing `defaultdict(lambda: Queue(maxsize=results_buffer_size or -1))`